pkg/artifacts/signable.go

```diff
@@ -293,7 +293,7 @@ func (oa *OCIArtifact) Signer(cfg config.Config) string {
 
 func (oa *OCIArtifact) Key(obj interface{}) string {
 	v := obj.(name.Digest)
-	return strings.TrimPrefix(v.DigestStr(), "sha256:")[:12]
+	return v.Name()
```

Comment:
We should double check the behavior here for annotation based storage - we should probably preserve the old behavior there since IIRC labels/annotations have a fixed key size, and including the full digest may cause issues.

Reply:
Great point!! Thanks for catching this @wlynch. You are right. I found the [source](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set).

Now, I maintained both ShortKey and FullKey so that downstream code can use whichever is needed. Please let me know what you think or if you have better suggestions.

Thank you!!